### DISCUSSION TITLE
runOnDemand WebRTC loading

### DISCUSSION THREAD

**Discussion by JoeHogan** [bluenviron/mediamtx#3327] [Questions]
## Describe the feature

Using `runOnDemand` works great but takes some time to begin sending WebRTC data to the client. In my case, the Alexa client closes the connection (times out) before runOnDemand can begin sending data... Since the client closes the connection, the FFMEG process is killed and you are now in a pickle. 

It would be great if the server could begin sending WebRTC data to the client prior to FFMPEG starting, as in, there is some kind of 'loading' track that you can push data from until FFMPEG is ready to begin sending data.

Obviously not sure how feasible this is...

**Comment by aler9** [comment]
Hello, it's not possible to send data to the client prior to the startup of FFmpeg, since it's not possible to know the list of tracks and their codecs prior to their publication by FFmpeg. My advice is to use `runOnInit` instead of `runOnDemand` in order to run FFmpeg continuously and create a stream that is immediately available.

[a comment by an automated account is not shown]
